### PR TITLE
fix: fix contract address display in confirmation page

### PR DESCRIPTION
**Pull request by sahar-fehri**
## **Description**

Sometimes the confirmation transaction page displays the destination wallet address rather than the contract address.

Here is related github issue: https://github.com/MetaMask/metamask-extension/issues/19502.

## **Manual testing steps**

_1. Go to : https://metamask.github.io/test-dapp/  or you can use etherscan on a contract address
_2. Deploy ERC1155 smart contract
_3. Call safeBatchTransferFrom function and notice the confirmation page does displays the destination address and not the contract address (side by side with the contract function name)

### **Before**

Calling mintBatch fct displays contract address correctly 

image

Calling safeBatchTransferFrom fct displays the destination address instead:

image

### **After**

Mint batch and safeTransferFromBatch display contract address

image

image

And if you are sending a transfer transaction (contractInteraction) to an address already imported in your wallet, it shows :

image

## **Related issues**

[_Fixes #MMAssets-33_](https://consensyssoftware.atlassian.net/browse/MMASSETS-33)

## **Pre-merge author checklist**

- [x ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ x] I've clearly explained:
  - [ ] What problem this PR is solving.
  - [ ] How this problem was solved.
  - [ ] How reviewers can test my changes.
- [x ] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ x] I’ve documented any added code.
- [ x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by pedronfigueiredo — DISMISSED** [review]

File: ui/pages/confirm-transaction-base/confirm-transaction-base.container.js [hunk 1]
```diff
@@ -83,8 +83,8 @@ import {
   TransactionStatus,
   TransactionType,
 } from '../../../shared/constants/transaction';
-import { getTokenAddressParam } from '../../helpers/utils/token-util';
 import { calcGasTotal } from '../../../shared/lib/transactions-controller-utils';
+import { getTokenAddressParam } from '../../helpers/utils/token-util';
```
Comment (brad-decker) [inline ui/pages/confirm-transaction-base/confirm-transaction-base.container.js:87]: Was this just a change in import order, if so was it intentional? if not i'd recommend trimming out this change to keep history a bit cleaner.

File: ui/components/app/confirm-page-container/confirm-page-container-content/confirm-page-container-summary/confirm-page-container-summary.component.js [hunk 2]
```diff
@@ -48,14 +51,15 @@ const ConfirmPageContainerSummary = (props) => {
...
+    // type of contract interaction it is "to" from txParams
+
     contractAddress =
       transactionType === TransactionType.tokenMethodTransfer ||
       transactionType === TransactionType.tokenMethodTransferFrom ||
       transactionType === TransactionType.tokenMethodSafeTransferFrom ||
       transactionType === TransactionType.tokenMethodSetApprovalForAll
         ? tokenAddress
-        : toAddress;
+        : txParamsToAddress;
```
[the remark this answers is not in the document]
Comment (brad-decker) [reply ui/components/app/confirm-page-container/confirm-page-container-content/confirm-page-container-summary/confirm-page-container-summary.component.js:62]: Is this only used in display? If so I think this is safe.

**Review by brad-decker — CHANGES_REQUESTED** [review]

File: ui/components/app/confirm-page-container/confirm-page-container-content/confirm-page-container-summary/confirm-page-container-summary.component.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (sahar-fehri) [reply ui/components/app/confirm-page-container/confirm-page-container-content/confirm-page-container-summary/confirm-page-container-summary.component.js:62]: Yes it is just for display

File: ui/pages/confirm-transaction-base/confirm-transaction-base.container.js (hunk 1 above)
Comment (sahar-fehri) [reply ui/pages/confirm-transaction-base/confirm-transaction-base.container.js]: That was not intentional! changing that 🙏

**Review by pedronfigueiredo — APPROVED** [review]

**Review by brad-decker — APPROVED** [review]

**Pull request merged**